In [35]:
import pandas as pd
import matplotlib.pyplot as plt 
%matplotlib inline 

In [36]:
housing_data = pd.read_csv("HW1/housing.csv")
housing_data.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0    -122.23     37.88                41.0        880.0           129.0   
1    -122.22     37.86                21.0       7099.0          1106.0   
2    -122.24     37.85                52.0       1467.0           190.0   
3    -122.25     37.85                52.0       1274.0           235.0   
4    -122.25     37.85                52.0       1627.0           280.0   

   population  households  median_income  median_house_value ocean_proximity  
0       322.0       126.0         8.3252            452600.0        NEAR BAY  
1      2401.0      1138.0         8.3014            358500.0        NEAR BAY  
2       496.0       177.0         7.2574            352100.0        NEAR BAY  
3       558.0       219.0         5.6431            341300.0        NEAR BAY  
4       565.0       259.0         3.8462            342200.0        NEAR BAY

### Q1- What is the data type of each feature? (ordinal/nominal/interval/ratio, discrete/continuous?

1. Longitude : **continuous**, as we see all are real numbers and are represented by float.
2. Latitude : **continuous** , same as latitude
3. Housing_median_age : **interval** 
4. Total_rooms : **discrete**
5. Total_bedrooms : **discrete**
6. Population : **discrete**
7. households : **discrete**
8. median_income : **continuos**. We see real values.
9. median_house_value : **discrete**. Countable set of values
10. ocean_proximity : **nominal**. It is divided into 5 categories

In [37]:
housing_data['housing_median_age'].unique()

array([41., 21., 52., 42., 50., 40., 49., 48., 51., 43.,  2., 46., 26.,
       20., 17., 36., 19., 23., 38., 35., 10., 16., 27., 39., 31., 29.,
       22., 37., 28., 34., 32., 47., 44., 30., 18., 45., 33., 24., 15.,
       14., 13., 25.,  5., 12.,  6.,  8.,  9.,  7.,  3.,  4., 11.,  1.])

### Q2- Display summary statistics of the data.  What can you learn from it on the data?

In [38]:
housing_data.describe()

longitude      latitude  housing_median_age   total_rooms  \
count  20640.000000  20640.000000        20640.000000  20640.000000   
mean    -119.569704     35.631861           28.639486   2635.763081   
std        2.003532      2.135952           12.585558   2181.615252   
min     -124.350000     32.540000            1.000000      2.000000   
25%     -121.800000     33.930000           18.000000   1447.750000   
50%     -118.490000     34.260000           29.000000   2127.000000   
75%     -118.010000     37.710000           37.000000   3148.000000   
max     -114.310000     41.950000           52.000000  39320.000000   

       total_bedrooms    population    households  median_income  \
count    20433.000000  20640.000000  20640.000000   20640.000000   
mean       537.870553   1425.476744    499.539680       3.870671   
std        421.385070   1132.462122    382.329753       1.899822   
min          1.000000      3.000000      1.000000       0.499900   
25%        296.000000    787.000000    280.000000       2.563400   
50%        435.000000   1166.000000    409.000000       3.534800   
75%        647.000000   1725.000000    605.000000       4.743250   
max       6445.000000  35682.000000   6082.000000      15.000100   

       median_house_value  
count        20640.000000  
mean        206855.816909  
std         115395.615874  
min          14999.000000  
25%         119600.000000  
50%         179700.000000  
75%         264725.000000  
max         500001.000000

**Observations:** 
1. Count gives the count of all the features. We see that the total bedrooms are shown as 20433 which means the summary is automatically ignoring the null values.
2. Min and Max are simply giving the min and max values of the features. And mean is giving the mean value of all numerical features.    
3. The summary stats are only for numerical features. So ocean_proximity is not included here.
4. Std is standard daviation and it shows how dispersed the values are.
5. 25%, 50%, and 75% rows show the corresponding percentiles. A percentile indicates the value below which a given percentage of observations in a group of observations fall. 
For eg, 25% of the districts have a housing_median_age lower than 18, while 50% are lower than 29 and 75% are lower than 37. 


In [32]:
# To verify the null values of Total_bedrooms
housing_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   longitude           20640 non-null  float64
 1   latitude            20640 non-null  float64
 2   housing_median_age  20640 non-null  float64
 3   total_rooms         20640 non-null  float64
 4   total_bedrooms      20433 non-null  float64
 5   population          20640 non-null  float64
 6   households          20640 non-null  float64
 7   median_income       20640 non-null  float64
 8   median_house_value  20640 non-null  float64
 9   ocean_proximity     20640 non-null  object 
dtypes: float64(9), object(1)
memory usage: 1.6+ MB


### Q3- Compute  the  correlation  between  each  feature  and  the  targetmedian_house_value.Which features have strong correlation with the target?